tornjak-backend/pkg/agent/auth/keycloak.go

```diff
@@ -0,0 +1,131 @@
+package auth
+
+import (
+	"os"
+	"fmt"
+	"strings"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/MicahParks/keyfunc"
+)
+
+type KeycloakVerifier struct {
+	jwks *keyfunc.JWKS
+}
+
+func NewKeycloakVerifier(jwksURL string) (*KeycloakVerifier) {
+	opts := keyfunc.Options{ // TODO add options to config file
+		RefreshErrorHandler: func(err error) {
+			fmt.Fprintf(os.Stdout, "error with jwt.Keyfunc: %v", err)
+		},
+		RefreshInterval:   time.Hour,
+		RefreshRateLimit:  time.Minute * 5,
+		RefreshTimeout:    time.Second * 10,
+		RefreshUnknownKID: true,
+	}
+	jwks, err := keyfunc.Get(jwksURL, opts)
+	if err != nil {
+		return nil
+	}
+	return &KeycloakVerifier {
+		jwks: jwks,
+	}
+}
+
+func get_token(r *http.Request) (string, error) {
+	// Authorization paramter from HTTP header
+	auth_header := r.Header.Get("Authorization")
```

Comment:
quick question, are there const to use for this?

Reply:
unfortunately I believe not